### PR TITLE
Setup clang format

### PR DESCRIPTION
**Pull request by jaykumarark**
I setup clang-format to format source files as a pre-build step for every target. This keeps the sources files neat and tidy.

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,44 @@
+﻿---
+BasedOnStyle: Google
```
Comment (jaykumarark) [inline .clang-format]: This is the style format rules which clang-format picks to format sources.

File: cmake-modules/phi-scripts.cmake [hunk 2]
```diff
@@ -0,0 +1,62 @@
+############################################
+# name: phi_add_exectuable
```
Comment (jaykumarark) [inline cmake-modules/phi-scripts.cmake]: This file will have cmake utility scripts for the project. Here I have added wrappers for add_executable and add_library function. These wrappers currently add clang formatting as a pre-build event for each target.

File: cmake-modules/phi-scripts.cmake [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    )
+    
+    if(PHI_WIN32)
+        add_executable(${PHI_TARGET} WIN32 ${PHI_SOURCES} ${PHI_UNPARSED_ARGUMENTS})
+    else()
+        add_executable(${PHI_TARGET} ${PHI_SOURCES} ${PHI_UNPARSED_ARGUMENTS})
+    endif()
+
+    # Run clang-format as pre-build event.    
+    add_custom_command(
```
Comment (jaykumarark) [inline cmake-modules/phi-scripts.cmake]: This basically adds the clang-format command to each target. All source file changes in this review are a result of running clang-format changes.

File: .clang-format (hunk 1 above)
Comment (varunk08) [reply .clang-format]: Whoa this is cool.

**Pull request merged**